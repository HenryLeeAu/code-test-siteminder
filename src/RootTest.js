// reuse redux provider in real component and testing code
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from 'reducers';
import ReduxThunk from 'redux-thunk';

export default ({ children, initialState = {} }) => {
  const store = createStore(
    reducers,
    initialState,
    applyMiddleware(ReduxThunk)
  );

  return <Provider store={store}>{children}</Provider>;
};
