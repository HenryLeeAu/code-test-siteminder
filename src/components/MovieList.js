import React, { Component } from 'react';
import MovieListItem from 'components/MovieListItem';
import { connect } from 'react-redux';
import { fetchMovieDetail } from 'actions';
class MovieList extends Component {
  handleClickList(id) {
    this.props.fetchMovieDetail(id);
  }
  renderList() {
    return this.props.searchStatus.currentMovieList.map((movie, index) => {
      return (
        <MovieListItem
          key={`${movie.imdbID}${movie.Title}`}
          onClick={e => this.handleClickList(movie.imdbID)}
          title={movie.Title}
          year={movie.Year}
          clicked={movie.imdbID === this.props.searchStatus.currentId}
        />
      );
    });
  }
  render() {
    return <ul className="movieList">{this.renderList()}</ul>;
  }
}
function mapStateToProps(state) {
  return {
    searchStatus: state.searchStatus,
  };
}

export default connect(
  mapStateToProps,
  { fetchMovieDetail }
)(MovieList);
