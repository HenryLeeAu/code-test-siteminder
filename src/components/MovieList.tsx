import React, { Component } from 'react';
import MovieListItem from './MovieListItem';
import { connect } from 'react-redux';
import { fetchMovieDetail } from '../actions';

interface IList {
  imdbID: string;
  Title: string;
  Year: string;
}
interface IProps {
  fetchMovieDetail(id: string): any;
  searchStatus: {
    currentMovieList: IList[];
    currentId: string;
  };
}
class MovieList extends Component<IProps> {
  handleClickList(id: string) {
    this.props.fetchMovieDetail(id);
  }
  renderList() {
    const { currentMovieList, currentId } = this.props.searchStatus;
    return currentMovieList.map((movie, index) => {
      const { imdbID, Title, Year } = movie;
      return (
        <MovieListItem
          key={`${imdbID}${Title}${index}`}
          onClick={(e: Event) => this.handleClickList(imdbID)}
          title={Title}
          year={Year}
          clicked={imdbID === currentId}
        />
      );
    });
  }
  render() {
    return (
      <ul className="movieList"  >
        {this.renderList()}
      </ul>
    );
  }
}
function mapStateToProps(state: any) {
  return {
    searchStatus: state.searchStatus,
  };
}

export default connect(
  mapStateToProps,
  { fetchMovieDetail }
)(MovieList);
