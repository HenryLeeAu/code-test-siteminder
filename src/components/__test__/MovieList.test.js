import React from "react";

import MovieList from "components/MovieList";
import { mount, unmount } from "enzyme";
import RootTest from "RootTest";

describe("MovieList", () => {
  let wrappered;
  const initialState = {
    searchStatus:{
      currentMovieList: [
        {
          imdbID: "id1",
          Title: "Marvel",
          Year: "2016"
        },
        {
          imdbID: "id2",
          Title: "Spider man",
          Year: "2016"
        }
      ]
    }
   
  };
  beforeEach(()=>{
    wrappered = mount(
      <RootTest initialState={initialState}>
        <MovieList />
      </RootTest>
    );
  })
  afterEach(()=>{
    wrappered.unmount();
  })
  it("render 2 movies", () => {
    expect(wrappered).toMatchSnapshot();
  })
  it('render 2 movies', ()=>{
    expect(wrappered.find(".movieList").find('li').length).toEqual(2)
   
  })
});
