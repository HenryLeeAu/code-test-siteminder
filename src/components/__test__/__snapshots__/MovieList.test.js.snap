// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList render UI 1`] = `
<_default
  initialState={
    Object {
      "searchStatus": Object {
        "currentMovieList": Array [
          Object {
            "Title": "Marvel",
            "Year": "2016",
            "imdbID": "id1",
          },
          Object {
            "Title": "Spider man",
            "Year": "2016",
            "imdbID": "id2",
          },
        ],
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(MovieList)>
      <MovieList
        fetchMovieDetail={[Function]}
        searchStatus={
          Object {
            "currentMovieList": Array [
              Object {
                "Title": "Marvel",
                "Year": "2016",
                "imdbID": "id1",
              },
              Object {
                "Title": "Spider man",
                "Year": "2016",
                "imdbID": "id2",
              },
            ],
          }
        }
      >
        <ul
          className="movieList"
        >
          <_default
            clicked={false}
            key="id1Marvel0"
            onClick={[Function]}
            title="Marvel"
            year="2016"
          >
            <li
              className=""
              onClick={[Function]}
            >
              <div>
                Marvel
              </div>
              <div
                className="align-right"
              >
                2016
              </div>
            </li>
          </_default>
          <_default
            clicked={false}
            key="id2Spider man1"
            onClick={[Function]}
            title="Spider man"
            year="2016"
          >
            <li
              className=""
              onClick={[Function]}
            >
              <div>
                Spider man
              </div>
              <div
                className="align-right"
              >
                2016
              </div>
            </li>
          </_default>
        </ul>
      </MovieList>
    </Connect(MovieList)>
  </Provider>
</_default>
`;
